### DISCUSSION TITLE
Why always get kernel dies when computing representations?

### DISCUSSION THREAD

**Discussion by Leo-T-Zang** [facebookresearch/esm#308] [Q&A]
Hi,
I am trying to use EMS2 to compute representation of some peptides in Jupyter Lab. But most of time I could only get results of 30 peptides, more than that, it shows kernel dies.
I use exactly the code provided
``
import torch
import esm
model, alphabet = esm.pretrained.esm2_t33_650M_UR50D()
batch_converter = alphabet.get_batch_converter()
model.eval()  # disables dropout for deterministic results

data = [
    ("protein1", "MKTVRQERLKSIVRILERSKEPVSGAQLAEELSVSRQVIVQDIAYLRSLGYNIVATPRGYVLAGG"),
    ("protein2", "KALTARQQEVFDLIRDHISQTGMPPTRAEIAQRLGFRSPNAAEEHLKALARKGVIEIVSGASRGIRLLQEE"),
   **data can only have 30 protein sequences, more than that kernel dies**
]
batch_labels, batch_strs, batch_tokens = batch_converter(data)

with torch.no_grad():
    results = model(batch_tokens, repr_layers=[33], return_contacts=True)
token_representations = results["representations"][33]

sequence_representations = []
for i, (_, seq) in enumerate(data):
    sequence_representations.append(token_representations[i, 1 : len(seq) + 1].mean(0))
``

**Comment by tomsercu** [comment]
Are you running on CPU or GPU? Your kernel will die if you run out of resources (probably memory). Maybe you're feeding a sequence that's too long?

**Reply by Leo-T-Zang** [reply]
Thank you for the reply. May I ask the max length of protein sequence that ESM2 can take?

**Reply by tomsercu** [reply]
Limit entirely dependent on the hardware and other details eg whether you use fp16 or fp32
